pkg/action/bootentries.go

```diff
@@ -0,0 +1,212 @@
+	systemdConf["default"] = entry
+	err = SystemdBootConfWriter(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"), systemdConf)
+	if err != nil {
+		cfg.Logger.Errorf("could not write loader.conf: %s", err)
+		return err
+	}
+	cfg.Logger.Infof("Default boot entry set to %s", entry)
+	return err
+}
+
+func ListBootEntries(cfg *config.Config) error {
+	// Get EFI partition
+	efiPartition, err := GetEfiPartition()
+	if err != nil {
+		return err
+	}
+	// Get default entry from loader.conf
+	loaderConf, err := utils.SystemdBootConfReader(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"))
+	if err != nil {
+		return err
+	}
+
+	entries, err := ListSystemdEntries(cfg, efiPartition)
+
+	// create a selector
+	selector := selection.New(fmt.Sprintf("Select Boot Entry (current entry: %s)", loaderConf["default"]), entries)
+	selector.Filter = nil        // Remove the filter
+	selector.ResultTemplate = `` // Do not print the result as we are asking for confirmation afterwards
+	selected, _ := selector.RunPrompt()
+	c := confirmation.New("Are you sure you want to change the boot entry to "+selected, confirmation.Yes)
+	c.ResultTemplate = ``
+	confirm, err := c.RunPrompt()
+	if confirm {
+		return SelectBootEntry(cfg, selected)
+	}
+	return err
+}
+
+// SystemdBootConfReader reads a systemd-boot conf file and returns a map with the key/value pairs
+func SystemdBootConfReader(filePath string) (map[string]string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	result := make(map[string]string)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, " ", 2)
+		if len(parts) == 2 {
+			result[parts[0]] = parts[1]
+		}
+		if len(parts) == 1 {
+			result[parts[0]] = ""
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// SystemdBootConfWriter writes a map to a systemd-boot conf file
+func SystemdBootConfWriter(filePath string, conf map[string]string) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	writer := bufio.NewWriter(file)
+	for k, v := range conf {
+		if v == "" {
+			_, err = writer.WriteString(fmt.Sprintf("%s \n", k))
+		} else {
+			_, err = writer.WriteString(fmt.Sprintf("%s %s\n", k, v))
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return writer.Flush()
+}
+
+func ListSystemdEntries(cfg *config.Config, efiPartition *v1.Partition) ([]string, error) {
+	var entries []string
+	err := fsutils.WalkDirFs(cfg.Fs, filepath.Join(efiPartition.MountPoint, "loader/entries/"), func(path string, info os.DirEntry, err error) error {
+		cfg.Logger.Debugf("Checking file %s", path)
+		if info.IsDir() {
+			return nil
+		}
+		if filepath.Ext(info.Name()) != ".conf" {
+			return nil
+		}
+		entries = append(entries, info.Name())
+		return err
```

Comment:
not making a comment to change, just to say that I find this a bit weird, since err is never set. Not sure if I prefer to have nil here to be clear that there's no chance of error, or if the variable needs to be used because of the compiler hehe

Reply:
nice, good catch xD

This should fail then? Where is this err coming from? I guess its currently the same as nil, but better make it explicit